rcgen/src/key_pair.rs

```diff
@@ -659,12 +542,9 @@ impl TryFrom<&PrivatePkcs8KeyDer<'_>> for KeyPair {
 	type Error = Error;
 
 	fn try_from(pkcs8: &PrivatePkcs8KeyDer) -> Result<KeyPair, Error> {
-		let (kind, alg) = KeyPair::from_pkcs8_raw(pkcs8)?;
-		Ok(KeyPair {
-			kind,
-			alg,
-			serialized_der: pkcs8.secret_pkcs8_der().into(),
-		})
+		let key = PrivateKeyDer::from(pkcs8);
```

Comment:
Nit: just skip the binding, `PrivateKeyDer::from(pkcs8).try_into()` should do it?